abmarl/sim/gridworld/base.py

```diff
@@ -40,6 +40,70 @@ def build_sim(cls, rows, cols, **kwargs):
 
         return cls._build_sim(rows, cols, **kwargs)
 
+    @classmethod
+    def build_sim_from_grid(cls, grid, **kwargs):
+        """
+        Build a GridSimluation from a Grid object.
+
+        Args:
+            grid: A Grid contains the all the agents index by location, so we can
+            construct a simluation from it.
+
+        Returns:
+            A GridSimulation built from the grid.
+        """
+        assert type(grid) is Grid, "Grid object required."
+        agents = {}
+        for r in range(grid.rows):
+            for c in range(grid.cols):
+                if grid[r, c] is not None:
+                    agents.update(grid[r, c])
+                    for agent in grid[r, c].values():
+                        np.testing.assert_array_equal(
+                            agent.initial_position,
+                            np.array([r, c])
+                        ), "The initial position of the agent must match its position in the grid."
```

Comment:
We could enforce that they match or we could update the agents' initial positions to match what is in the grid. Depends on how the builder is used. Leave it as is for now, and update it later if needed.